.github/workflows/ci.yaml

```diff
@@ -0,0 +1,68 @@
+name: CI
+on: [push]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    env:
+      DJANGO_SETTINGS_MODULE: rca.settings.test
+      SECRET_KEY: testingsecretkey
+      SECURE_SSL_REDIRECT: false
+      ALLOWED_HOSTS: localhost
+      DATABASE_URL: postgres://postgres:postgres@localhost/postgres
+      BIRDBATH_REQUIRED: false
+
+    services:
+      postgres:
+        image: postgres:11.5
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: postgres
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - name: Install npm dependencies
+        run: npm install
+      - name: CSS linting
+        run: npm run lint:css
+      - name: JS linting
+        run: npm run lint:js
+      - name: Prettier
+        run: npm run format
+      - name: JS tests
+        run: npm run test
+      - name: Compile static files
+        run: npm run build:prod
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+      - uses: Gr1N/setup-poetry@v8
+        with:
+          poetry-version: 1.5.0
+      - name: Install Poetry
+        run: |
+          poetry config virtualenvs.create false &&
+          poetry install
+      # - name: Flake8
+      #   run: flake8 ./rca fabfile.py
```

Comment:
I think we need flake 8 to run.

As a sidenote, and something I think we should do in CI, you can run the entire pre-commit suite via:

```yml
jobs:
  precommit:
    name: Run precommit
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - name: Set up Python
        uses: actions/setup-python@v4
        with:
          python-version: 3.11

      - name: Set up Node.js
        uses: actions/setup-node@v4
        with:
          node-version: 20

      - run: npm ci --no-optional --no-audit --progress=false

      - name: Install pre-commit
        run: pip install pre-commit

      - name: Cache pre-commit dependencies
        uses: actions/cache@v3
        with:
          path: |
            ~/.cache/pre-commit
          key: pre-commit-${{ hashFiles('.pre-commit-config.yaml') }}

      - name: Run pre-commit
        run: pre-commit run --all-files
```

Reply:
Ah precommit is another area I am not too familiar with. I thought it was something that ran on your local machine, with certain tests that have to pass before you can commit code, rather than something that runs in CI? It's something I once tried years ago but haven't used since. Is it worth trying to set it up on my machine so I understand how it works?

Reply:
It _is_ something that runs on your local machine yea, I'd highly recommend trying it out locally.
We are starting to use `pre-commit` in the CI too just because it ensures the following:
- All linting and formatting rules/versions are contained in pre-commit-config.yml
- It keeps the ci script a bit smaller when it comes to formatting and lining, we just just run the single pre-commit task.

It's not really a prerequestie of this PR but it's worth being aware of that as we are starting to bring it in more

Reply:
Are there some docs I can read somewhere on how to test it out?